android/src/main/java/com/polidea/reactnativeble/BleModule.java

```diff
@@ -1038,33 +1041,46 @@ private void safeMonitorCharacteristicForDevice(final Characteristic characteris
-                .flatMap(new Func1<RxBleConnection, Observable<Observable<byte[]>>>() {
-                    @Override
-                    public Observable<Observable<byte[]>> call(RxBleConnection connection) {
-                        int properties = gattCharacteristic.getProperties();
-                        if ((properties & BluetoothGattCharacteristic.PROPERTY_NOTIFY) != 0) {
-                            return connection.setupNotification(gattCharacteristic);
-                        }
+        final int properties = gattCharacteristic.getProperties();
+        final boolean notifications = (properties & BluetoothGattCharacteristic.PROPERTY_NOTIFY) != 0;
+        final boolean indications = (properties & BluetoothGattCharacteristic.PROPERTY_INDICATE) != 0;
 
-                        if ((properties & BluetoothGattCharacteristic.PROPERTY_INDICATE) != 0) {
-                            return connection.setupIndication(gattCharacteristic);
-                        }
+        final BluetoothGattDescriptor cccDescriptor =
+                gattCharacteristic.getDescriptor(Characteristic.CLIENT_CHARACTERISTIC_CONFIG_UUID);
 
-                        return Observable.error(new CannotMonitorCharacteristicException(gattCharacteristic));
-                    }
-                })
+        final byte[] enableValue = notifications
+                ? BluetoothGattDescriptor.ENABLE_NOTIFICATION_VALUE
+                : BluetoothGattDescriptor.ENABLE_INDICATION_VALUE;
+        final byte[] disableValue = BluetoothGattDescriptor.DISABLE_NOTIFICATION_VALUE;
+
+        Observable<Observable<byte[]>> notificationCompatObservable = notifications || indications
+                // NotificationSetupMode.COMPAT does not write CCC Descriptor on it's own
+                ? connection.setupNotification(gattCharacteristic, NotificationSetupMode.COMPAT)
+                : Observable.<Observable<byte[]>>error(new CannotMonitorCharacteristicException(gattCharacteristic));
+
+        final Observable<byte[]> enableNotificationObservable = cccDescriptor != null
+                ? connection.writeDescriptor(cccDescriptor, enableValue)
+                : Observable.<byte[]>empty();
+
+        final Observable<byte[]> disableNotificationObservable = cccDescriptor != null
+                ? connection.writeDescriptor(cccDescriptor, disableValue)
+                : Observable.<byte[]>empty();
+
+        final Subscription subscription = notificationCompatObservable
                 .flatMap(new Func1<Observable<byte[]>, Observable<byte[]>>() {
                     @Override
                     public Observable<byte[]> call(Observable<byte[]> observable) {
-                        return observable;
+                        // Keep in mind that every subscription to this observable will initiate another descriptor write
+                        return observable.mergeWith(enableNotificationObservable.ignoreElements());
```

Comment:
I don't want to register for notification every call. Only first registration should send BLE packet.